### PR TITLE
Feature/whitelist-review-frontend

### PR DESCRIPTION
**Pull request by OliverNeff**
## Description:
Showing whitelist panel just in whitelist mode of course add some optical improvements and make better snackbar infos at creating whitelist users.

## Known Issues:
Review at 2018-01-18
see #501
------
_Remember to prefix your PR-Title with `⚠ WIP: ` if you still work on it.
If you have reached a final state, remove the prefix._

### CODE REVIEW

File: app/webFrontend/src/app/course/course-edit/members/whitelist-edit/whitelist-edit.component.ts [hunk 1]
```diff
@@ -62,21 +60,34 @@ export class WhitelistEditComponent implements OnInit {
       .map(val => val ? this.filter(val, this.uidOptions) : this.uidOptions.slice());
   }
 
-  createNewWhitelistUser() {
+  async createNewWhitelistUser() {
     if (this.whitelistUser.firstName === '') {
-      this.snackBar.open('First name should not be empty.', '', {duration: 6000});
+      this.snackBar.open('First name should not be empty.', '', {duration: this.snackBarDuration});
```
Comment (PatrickSkowronek) [inline app/webFrontend/src/app/course/course-edit/members/whitelist-edit/whitelist-edit.component.ts]: const correctness: I would expect a dismiss button

File: app/webFrontend/src/app/course/course-edit/members/whitelist-edit/whitelist-edit.component.ts [hunk 2]
```diff
@@ -62,21 +60,34 @@ export class WhitelistEditComponent implements OnInit {
...
       return null;
     } else if (this.whitelistUser.lastName === '') {
-      this.snackBar.open('Last name should not be empty.', '', {duration: 6000});
+      this.snackBar.open('Last name should not be empty.', '', {duration: this.snackBarDuration});
```
Comment (PatrickSkowronek) [inline app/webFrontend/src/app/course/course-edit/members/whitelist-edit/whitelist-edit.component.ts]: const correctness: I would expect a dismiss button

File: app/webFrontend/src/app/course/course-edit/members/whitelist-edit/whitelist-edit.component.ts [hunk 3]
```diff
@@ -62,21 +60,34 @@ export class WhitelistEditComponent implements OnInit {
...
       return null;
     } else if (this.whitelistUser.uid === '' || isNaN(Number(this.whitelistUser.uid))) {
-      this.snackBar.open('Unique identification should be a number and not empty', '', {duration: 6000});
+      this.snackBar.open('Unique identification should be a number and not empty', '', {duration: this.snackBarDuration});
```
Comment (PatrickSkowronek) [inline app/webFrontend/src/app/course/course-edit/members/whitelist-edit/whitelist-edit.component.ts]: const correctness: I would expect a dismiss button

File: app/webFrontend/src/app/course/course-edit/members/whitelist-edit/whitelist-edit.component.ts [hunk 4]
```diff
@@ -62,21 +60,34 @@ export class WhitelistEditComponent implements OnInit {
...
     }
     this.whitelistUser.courseId = this.course._id;
-    return this.whitelistUserService.createItem(this.whitelistUser).then((newUser) => {
+    try {
+      const newUser: IWhitelistUser = await this.whitelistUserService.createItem(this.whitelistUser);
       this.dragableWhitelistUser.push(newUser);
       this.onDragendPushWhitelist.emit(newUser);
-    });
+      if (this.showComponent.isInCourse(newUser)) {
+        this.snackBar.open('Create whitelist user. User is in course.', '', {duration: this.snackBarDuration});
```
Comment (PatrickSkowronek) [inline app/webFrontend/src/app/course/course-edit/members/whitelist-edit/whitelist-edit.component.ts]: const correctness: I would expect a dismiss button

File: app/webFrontend/src/app/course/course-edit/members/whitelist-edit/whitelist-edit.component.ts [hunk 5]
```diff
@@ -62,21 +60,34 @@ export class WhitelistEditComponent implements OnInit {
...
+      } else if (this.showComponent.isSignedIn(newUser)) {
+        this.snackBar.open('Create whitelist user. User is signed in.', '', {duration: this.snackBarDuration});
```
Comment (PatrickSkowronek) [inline app/webFrontend/src/app/course/course-edit/members/whitelist-edit/whitelist-edit.component.ts]: const correctness: I would expect a dismiss button

File: app/webFrontend/src/app/course/course-edit/members/whitelist-edit/whitelist-edit.component.ts [hunk 6]
```diff
@@ -62,21 +60,34 @@ export class WhitelistEditComponent implements OnInit {
...
+      } else {
+        this.snackBar.open('Create whitelist user. User is not signed in.', '', {duration: this.snackBarDuration});
```
Comment (PatrickSkowronek) [inline app/webFrontend/src/app/course/course-edit/members/whitelist-edit/whitelist-edit.component.ts]: const correctness: I would expect a dismiss button

File: app/webFrontend/src/app/course/course-edit/members/whitelist-edit/whitelist-edit.component.ts [hunk 7]
```diff
@@ -62,21 +60,34 @@ export class WhitelistEditComponent implements OnInit {
...
+      }
+      return newUser;
+    } catch (exception) {
+      this.snackBar.open('Could not create whitelist user. Error was: '
```
Comment (PatrickSkowronek) [inline app/webFrontend/src/app/course/course-edit/members/whitelist-edit/whitelist-edit.component.ts]: const correctness: I would expect a dismiss button

**Comment by PatrickSkowronek** [comment]
Bug: I can add the same student over and over again:

screenshot from 2018-01-22 22-44-25

This will be saved multiple times in the database!

**Comment by OliverNeff** [comment]
see: https://github.com/h-da/geli/pull/493

**Comment by OliverNeff** [comment]
Explanation for outdated: 
this.duration is now a static readonly
Dismiss button will be another ticket, because we should integrate our drop down uploader. 
But I think it is a little buggy at this time.

**Comment by PatrickSkowronek** [comment]
Adding to each commented this line 
` this.snackBar.open(t['auth.loginSuccess'], t['common.dismiss'], {duration: 2000});`
is hopefully not that time consuming.

File: api/src/controllers/UserController.ts [hunk 8]
```diff
@@ -68,8 +68,7 @@ export class UserController {
       'score': {$meta: 'textScore'}
     })
       .where({role: role})
-      .sort({'score': {$meta: 'textScore'}})
-      .limit(20);
```
Comment (lukaskorte) [inline api/src/controllers/UserController.ts]: Why remove the limit?

Comment (OliverNeff) [reply api/src/controllers/UserController.ts]: Problems by checking a user if it is in system when there are more than 20.

Comment (OliverNeff) [reply api/src/controllers/UserController.ts]: It would maybe better when you can set it over a parameter.

**Comment by OliverNeff** [comment]
Closed because of #501

**Pull request closed without merging**